File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let hashMap = new Map();
    
    participant.forEach((x) => {
        if(hashMap.has(x)) hashMap.set(x, hashMap.get(x)+1);
        else hashMap.set(x, 1);
    });
    
    completion.forEach((x) => {
        hashMap.set(x, hashMap.get(x)-1);
    });
    
    for(let [key, value] of hashMap) {
        if(value===1) return key;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dict = {}

    for i, person in enumerate(participant):
        dict[person] = dict.get(person, 0) + 1
        if i < len(completion):
            dict[completion[i]] = dict.get(completion[i], 0) - 1

    for key, value in dict.items():
        if value > 0:
            return key
